resolwe/permissions/utils.py

```diff
@@ -21,18 +21,61 @@
 from rest_framework import exceptions
 
 
+def get_perm_action(perm):
+    """Split action from permission of format.
+
+    Input permission can be in the form of '<action>_<object_type>'
+    or '<action>'.
+    """
+    return perm.split('_', 1)[0]
+
+
+def get_full_perm(perm, obj):
+    """Join action with the content type of ``obj``.
+
+    Permission is returned in the format of ``<action>_<object_type>``.
+    """
+    ctype = ContentType.objects.get_for_model(obj)
+    # Camel case class names are converted into a space-separated
+    # content types, so spaces have to be removed.
+    ctype = str(ctype).replace(' ', '')
+
+    return '{}_{}'.format(perm.lower(), ctype)
+
+
+def get_all_perms(obj):
+    """Return a list of all permissions on ``obj``."""
+    return list(zip(*obj._meta.permissions))[0]  # pylint: disable=protected-access
+
+
+def change_perm_ctype(perm, dest_obj):
+    """Keep permission action and change content type to ``dest_obj``."""
+    action = get_perm_action(perm)
+    return get_full_perm(action, dest_obj)
+
+
 def copy_permissions(src_obj, dest_obj):
     """Copy permissions form ``src_obj`` to ``dest_obj``."""
+    def _process_permission(codename, user_or_group, dest_obj):
+        """Process single permission."""
+        if relabel:
```

Comment:
Does this implicitly come from the parent scope? If so, this should be changed, so that either `relabel` is defined above this function or (better) that relabel is a parameter.